src/Compilers/CSharp/Test/Semantic/Semantics/SynthesizedStaticConstructorTests.cs

```diff
@@ -265,6 +265,24 @@ class C
             Assert.True(IsBeforeFieldInit(typeSymbol));
         }
 
+        [WorkItem(543606, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/543606")]
+        [Fact]
+        public void StaticConstructorNullInitializer()
+        {
+            var source = @"
+#nullable enable
+class C
+{
+    static string s1 = null!;
+}";
+
+            var typeSymbol = CompileAndExtractTypeSymbol(source);
+
+            // Although we do not emit the synthesized static constructor, the source type symbol will still appear to have one
+            Assert.True(HasSynthesizedStaticConstructor(typeSymbol));
+            Assert.True(IsBeforeFieldInit(typeSymbol));
+        }
```

Comment:
>} [](start = 8, length = 1)

Please test:
```
static S s1 = default;
static object s2 = default(S);
```

Reply:
Consider using a `Theory` to keep the test compact.